### PR TITLE
chore(dev-infra): create script to generate (and check) library resources

### PR DESCRIPTION
**Pull request by Blackbaud-SteveBrush**

### CODE REVIEW

File: scripts/utils/git-utils.ts [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+  };
+
+  if (config) {
+    options = { ...options, ...config };
+  }
+
+  let isClean = result.includes('nothing to commit, working tree clean');
+
+  if (isClean && options.compareAgainstRemote) {
+    isClean = result.includes('Your branch is up to date');
```
Comment (johnhwhite) [inline scripts/utils/git-utils.ts:25]: This needs a `git fetch` so the local information is up to date.

File: libs/components/ag-grid/src/lib/modules/shared/sky-ag-grid-resources.module.ts [hunk 2]
```diff
@@ -32,17 +32,23 @@ const RESOURCES: { [locale: string]: SkyLibResources } = {
       message: 'Editable currency {0} for row {1}',
     },
     sky_ag_grid_cell_renderer_number_validator_message: {
-      message: 'Please enter a valid number',
+      message: 'Enter a valid number.',
     },
     sky_ag_grid_cell_renderer_currency_validator_message: {
-      message: 'Please enter a valid currency',
+      message: 'Enter a valid currency.',
```
Comment (johnhwhite) [inline libs/components/ag-grid/src/lib/modules/shared/sky-ag-grid-resources.module.ts:38]: Are these changes intentional or for testing?

Comment (Blackbaud-SteveBrush) [reply libs/components/ag-grid/src/lib/modules/shared/sky-ag-grid-resources.module.ts:38]: All of the `*-resources.module.ts` file changes are a result of the script fixing/updating their contents. So in this case, we had updated the resource JSON, but never ran `ng g @skyux/i18n:lib-resources-module` on `ag-grid` before releasing.

Comment (Blackbaud-SteveBrush) [reply libs/components/ag-grid/src/lib/modules/shared/sky-ag-grid-resources.module.ts:38]: In short, this change is intentional.

File: scripts/utils/git-utils.ts (hunk 1 above)
Comment (Blackbaud-SteveBrush) [reply scripts/utils/git-utils.ts:25]: Good catch! Done.

File: libs/components/a11y/src/lib/modules/shared/sky-a11y-resources.module.ts [hunk 3]
```diff
@@ -19,7 +19,10 @@ const RESOURCES: { [locale: string]: SkyLibResources } = {
 };
 
 export class SkyA11yResourcesProvider implements SkyLibResourcesProvider {
-  public getString(localeInfo: SkyAppLocaleInfo, name: string): string {
+  public getString(
+    localeInfo: SkyAppLocaleInfo,
+    name: string
+  ): string | undefined {
```
Comment (Blackbaud-SteveBrush) [inline libs/components/a11y/src/lib/modules/shared/sky-a11y-resources.module.ts:25]: ☝️ Strict-mode fixes (auto-generated by the script).

File: libs/components/a11y/src/lib/modules/shared/sky-a11y-resources.module.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
 /**
  * NOTICE: DO NOT MODIFY THIS FILE!
  * The contents of this file were automatically generated by
- * the 'ng generate @skyux/i18n:lib-resources-module modules/shared/sky-a11y' schematic.
+ * the 'ng generate @skyux/i18n:lib-resources-module lib/modules/shared/sky-a11y' schematic.
```
Comment (Blackbaud-SteveBrush) [inline libs/components/a11y/src/lib/modules/shared/sky-a11y-resources.module.ts:4]: ☝️ This comment is auto-generated by the `@skyux/i18n` schematic, but is a bit useless to us since `ng g @skyux/i18n:lib-resources-module` can't be executed within the context of the monorepo without building the library and `npm link`ing it.

File: libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.module.ts [hunk 5]
```diff
@@ -2,11 +2,11 @@ import { NgModule } from '@angular/core';
 import { FormsModule } from '@angular/forms';
 import { SkyAutocompleteModule } from '@skyux/lookup';
 
-import { AgGridResourcesModule } from '../../../shared/ag-grid-resources.module';
+import { SkyAgGridResourcesModule } from '../../../shared/sky-ag-grid-resources.module';
 import { SkyAgGridCellEditorAutocompleteComponent } from '../cell-editor-autocomplete/cell-editor-autocomplete.component';
 
 @NgModule({
-  imports: [FormsModule, AgGridResourcesModule, SkyAutocompleteModule],
+  imports: [FormsModule, SkyAgGridResourcesModule, SkyAutocompleteModule],
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/modules/ag-grid/cell-editors/cell-editor-autocomplete/cell-editor-autocomplete.module.ts:9]: The prefix `Sky-` changes are just to make every library consistent and easier to regen the library resources in one command.

**Review by Blackbaud-SteveBrush — COMMENTED** [review]
The `npm run dev:create-library-resources` script:
1. builds the `@skyux/i18n` library
2. runs the `lib-resources-module` schematic from the `dist` folder for each library that has a non-empty resources JSON file.

**Review by johnhwhite — APPROVED** [review]
